### DISCUSSION TITLE
[Idea] Add to TheXTech and MoonDust - Editor Discord Rich Presence

### DISCUSSION THREAD

**Discussion by SkullGamer205** [Wohlstand/TheXTech#507] [Ideas]
# **Prelude**

Today, such a small trinket for TheXTech and MoonDust - Editor as Discord Rich Presence came to mind. As it turns out, I wasn't the only one thinking about this. In principle, this is a trinket, but it could be appropriate in modern realities. _If you want to share what you play, that would be helpful..._

# **Idea**

- [x] **Implement Discord Rich Presence in TheXTech and MoonDust - Editor**

# **Implementation of the idea**

### For TheXTech:

1. **Name of Asset Pack** {Super Mario Bros. X/A2XT/Nostalgic Paradise, e.t.c.} (Read from `gameinfo.ini` in compartment `[game]`)
2. **Status** {Playing a Level|World / Creating a Level|World / in Menu}
3. **Name Of World/Level**
4. **Time** (This time is taken from the world, not from the moment the game was turned on, so in the **In Menu** state, the time should be hidden, as well as the name of the world)

##### Variant 1 (SMB.X):
![2023-04-01--19-15-08](https://user-images.githubusercontent.com/61167080/229277647-cb918334-027e-4f04-9ea0-54be665ac1bc.png)

##### Variant 2 (SMB.X) (This Variant is better):
![2023-04-01--19-16-53](https://user-images.githubusercontent.com/61167080/229277765-2d98ed46-be89-4445-9b6e-008602c45d54.png)

##### Variant 1 (A2XT):
![2023-04-01--19-20-37](https://user-images.githubusercontent.com/61167080/229277710-3112d1fa-7a01-4b6e-a1e3-8373873e1b33.png)

##### Variant 2 (A2XT) (Better):
![2023-04-01--19-19-59](https://user-images.githubusercontent.com/61167080/229277885-1c661d19-b99d-4a87-9e1b-52652d953f0d.png)

_Important note: The picture with the blue mushroom will either be shipped **TOGETHER** with the rest of the icons in `/graphics/ui/icon/` `(Where the status icons will be downloaded from, by the way)` or will be implanted somewhere._

### For MoonDust - Editor

1. **Name Of Programm**
2. **Status** {Editing a Level|World}
3. **Name Of World/Level**
4. **Time**

#### Example in picture:

![2023-04-01--19-11-30](https://user-images.githubusercontent.com/61167080/229277974-444763f2-ec45-4e08-9202-4aeb5599f424.png)

**_If you have any ideas, please suggest._**

**Comment by Wohlstand** [comment]
On the TheXTech side, the thing had been completely implemented.
